Ensure inputs to tensor backend einsum test are tensors

matthewfeickert opened the pull request:
# Description

Resolves #482 

In TensorFlow `v1.14` [`tf.einsum`](https://www.tensorflow.org/api_docs/python/tf/einsum) is expecting Tensors. As such, ensure that in the `test_einsum` all inputs are in fact Tensors.

# Checklist Before Requesting Reviewer

- [x] Tests are passing
- [x] "WIP" removed from the title of the pull request
- [x] Selected an Assignee for the PR to be responsible for the log summary

# Before Merging

For the PR Assignees:

- [x] Summarize commit messages into a comprehensive review of the PR

```
- Ensure that all inputs to the test_einsum test are Tensors
- Update minimum compatible version of TensorFlow to v1.14 to avoid future compatibility issues
```

kratsg commented:
We should be fixing the test, not the code. This will slow done the actual einsum calculation in tensorflow by doing the tensor check/conversion every time it's called, no?

matthewfeickert commented:
> We should be fixing the test, not the code. This will slow done the actual einsum calculation in tensorflow by doing the tensor check/conversion every time it's called, no?

Yes, this was the conclusion I also came to while I was thinking about this in the shower. I've corrected this and rebased the PR.
